lib/auth/recordingmetadata/recordingmetadatav1/membuffer.go

```diff
@@ -0,0 +1,55 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package recordingmetadatav1
+
+import (
+	"sync"
+
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+)
+
+// memBuffer is an in-memory byte buffer that implements both io.Writer and
```

Comment:
It's not too clear to me what the necessity of this is - it doesn't look like manager.WriteAtBuffer is accessed concurrently in the service itself.

Reply:
removed the mutex